src/tensor_ops/matmul/cpu_kernel.rs

```diff
@@ -71,14 +71,14 @@ impl MatMulImpl<half::f16> for Cpu {
                 m.size(),
                 n.size(),
                 k.size(),
-                cp as *mut gemm::f16,
+                cp,
                 cstr[1] as isize,
                 cstr[0] as isize,
                 accum,
-                ap as *const gemm::f16,
+                ap,
                 astr[1] as isize,
                 astr[0] as isize,
-                bp as *const gemm::f16,
+                bp,
```

Comment:
We still need these because gemm might use a different version of f16. Internally gemm does a TypeId check against it's own version of f16, and the same type from two different versions of crates will have different TypeId